### ISSUE TITLE
Site freezes after selection and clearing on Autocomplete demo 

### ISSUE THREAD

**Issue by bradlygreen** [angular/material#3800]
To reproduce on Chrome 43:
1. Go to https://material.angularjs.org/latest/#/demo/material.components.autocomplete
2. Select a state
3. Click the 'X' to clear the state
4. Try to click on anything else on the page.  Observe the unresponsive status.

**Comment by ThomasBurleson** [comment]
@bradlygreen - Rob and I are are seeing something similar for Select menus... working on a fix today; seems to be present only with Angular 1.4.x.

**Comment by ThomasBurleson** [comment]
> Occassionally the transitionEnd event will not fire and the '$mdUtil. waitTransitionEnd()' process will timeout; which rejects the pending promise. Using '.finally()' [consistently] allows the correct handlers to fire and will continuing to propagate the rejected promise.

**Comment by linkedrank** [comment]
Hello - is there progress on this? I am also experiencing this issue.

**Comment by topherfangio** [comment]
@linkedrank Per the close comment above, this was fixed in d6a938e983690c229c1ee856c0416ae1832cffb4. Are you still seeing this is you use the latest master code?

**Comment by linkedrank** [comment]
Apologies - somehow I missed the fact that it was closed.

I am still seeing it. I'm using the autocomplete in combination with a
grid. See this screencast: http://screencast.com/t/fItdc55Ae4VI.

One thing I am doing though is the below to hide the autocomplete on the
blur of the underlying input element. It's not ng-blur, so I end up having
to call $scope.apply, which could be I'm sure introducing something funny.
This blur event goes on to hide the autocomplete widget. However, this blur
tactic was working fine for the past week, and I've only now started seeing
this behavior. I updated to do AngularJS 1.4.5 yesterday, so I'm not sure
if that's the cause.


        if (scope.hideOnBlur) {
          setTimeout(function() {

document.getElementById(scope.autocompleteId).addEventListener('blur',
scope.onblur);
          }, 0);
        }
